jgiven-tests/src/test/java/com/tngtech/jgiven/impl/GuaranteedStateTest.java

```diff
@@ -0,0 +1,54 @@
+package com.tngtech.jgiven.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.tngtech.jgiven.Stage;
+import com.tngtech.jgiven.exception.JGivenMissingGuaranteedScenarioStateException;
+import com.tngtech.jgiven.junit.SimpleScenarioTest;
+import com.tngtech.jgiven.report.model.ReportModel;
+import com.tngtech.jgiven.testframework.TestExecutor;
+import com.tngtech.jgiven.testframework.TestFramework;
+import com.tngtech.jgiven.tests.GuaranteedFieldRealTest;
+import com.tngtech.jgiven.tests.TestScenarioRepository;
+import org.junit.Test;
+
+public class GuaranteedStateTest extends SimpleScenarioTest<GuaranteedStateTest.SimpleTestStage> {
+
+    @Test
+    public void assure_before_method_of_second_test_is_executed_after_guaranteed_fields_validation() {
+        given().a_Jgiven_test_with_a_guaranteed_null_state();
+        when().the_test_is_executed();
+        then().the_report_contains_$_exception(JGivenMissingGuaranteedScenarioStateException.class);
+    }
+
+    @Test
+    public void assure_before_method_of_second_test_is_executed_if_guaranteed_initialized() {
+        given().a_Jgiven_test_with_a_guaranteed_state();
+        when().the_test_is_executed();
+        then().the_report_contains_$_exception(ClassNotFoundException.class);
+    }
+
+    public static class SimpleTestStage extends Stage<SimpleTestStage> {
+        TestScenarioRepository.TestScenario testScenario;
+        private ReportModel testReport;
+
+        public void a_Jgiven_test_with_a_guaranteed_null_state() {
+            testScenario = new TestScenarioRepository.TestScenario(GuaranteedFieldRealTest.class, "a_sample_test");
+        }
+
+        public void a_Jgiven_test_with_a_guaranteed_state() {
+            testScenario = new TestScenarioRepository.TestScenario(GuaranteedFieldRealTest.class, "a_sample_initialized_test");
+        }
+
+        public void the_test_is_executed() {
+            testReport = TestExecutor.getExecutor(TestFramework.JUnit)
```

Comment:
technically fine, but I think it would be nicer to read, if you separated that into two statements. As a side effect, if one of those methods throws an unexpected exception, it will be easier to figure out which of the two threw it. 
(its the avoid train-wreck rule :))